Add information about caching and verifications in "Load a Dataset" docs (#1705)
huggingface/datasets: docs/source/loading_datasets.rst

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

You can control the way the the :func:`datasets.load_dataset` function handles already downloaded data by setting its :obj:`download_mode` parameter.

By default, :obj:`download_mode` is set to ``"reuse_dataset_if_exists"``. The :func:`datasets.load_dataset` function will reuse both raw downloads and the prepared dataset, if they exist in the cache directory.

The following table describes the three available modes for download:

.. list-table:: Behavior of :func:`datasets.load_dataset` depending on :obj:`download_mode`
   :header-rows: 1

   * - :obj:`download_mode` parameter value
     - Downloaded files (raw data)
     - Dataset object
   * - ``"reuse_dataset_if_exists"`` (default)
     - Reuse
     - Reuse
   * - ``"reuse_cache_if_exists"``
     - Reuse
     - Fresh
   * - ``"force_redownload"``  
     - Fresh
     - Fresh

For example, you can run the following if you want to force the re-download of the SQuAD raw data files:

.. code-block::

    >>> from datasets import load_dataset
    >>> dataset = load_dataset('squad', download_mode="force_redownload")

Integrity verifications
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> When downlaoding a dataset from the 🤗 dataset hub, the :func:`datasets.load_dataset` function performs by default a number of verifications on the downloaded files. These verifications include:

Reviewer: ```suggestion
When downloading a dataset from the 🤗 dataset hub, the :func:`datasets.load_dataset` function performs by default a number of verifications on the downloaded files. These verifications include:
```

> - Verifying the number and size in bytes of the downloaded files

Reviewer: ```suggestion
- Verifying the list of downloaded files
- Verifying the number of bytes of the downloaded files
```

- Verifying the SHA256 checksums of the downloaded files
- Verifying the number of splits in the generated `DatasetDict`
- Verifying the number of samples in each split of the generated `DatasetDict`

You can disable these verifications by setting the :obj:`ignore_verifications` parameter to ``True``.

> You have the possibility to locally override the informations used to perform the integrity verifications by setting the :obj:`save_infos` parameter to ``True``.

Reviewer: ```suggestion
You also have the possibility to locally override the informations used to perform the integrity verifications by setting the :obj:`save_infos` parameter to ``True``.
```
